lib/gitlab/ci/templates/jobs/docker/Kaniko.yml

```diff
@@ -51,3 +53,48 @@ kaniko_publish:
     - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
     - if: $CI_PIPELINE_SOURCE == "web"
     - if: $CI_PIPELINE_SOURCE == "merge_request_event"
+
+# kaniko_publish_arm: Extends the standard Kaniko publish job for ARM-based architectures.
+# This job inherits all logic from `.kaniko_publish` and customizes execution by using a specific GitLab runner tag.
+# The tag `architecture-aarch64` ensures that the job runs on an ARM64 (AArch64) compatible GitLab runner.
+#
+# Use Case:
+# - When building Docker images for ARM-based environments such as AWS Graviton, Apple M1/M2 chips, or Jetson devices.
+# - Ensures proper architecture targeting during multi-arch image builds or ARM-specific deployments.
+kaniko_publish_arm:
+  stage: publish
+  extends: .kaniko_publish
+  tags:
+    - architecture-aarch64
+  rules:
+    - when: always
+
+# Standalone multiarch manifest job with its own variables
+multiarch_manifest_publish:
+  stage: publish
+  image: docker:24.0.5
+  services:
+    - docker:dind
+  variables:
+    DOCKER_TLS_CERTDIR: ""
+    AMD_IMAGE: "harbor.ctic-dev.com/platform-engineering/getting-started:0.1.2"
+    ARM_IMAGE: "harbor.ctic-dev.com/platform-engineering/getting-started:0.1.5"
+    MULTIARCH_IMAGE: "harbor.ctic-dev.com/platform-engineering/getting-started:0.1.5-multiarch"
+    HARBOR_URL: "https://harbor.ctic-dev.com"
```

Comment:
Instead of HARBOR_URL, can you use Gitlab CI variables we already use with these template? I think DOCKER_REPO_HOSTNAME (e.g., from https://github.com/chesapeaketechnology/gitlab-templates/blob/release/4/lib/gitlab/ci/templates/jobs/docker/Img.yml#L6 and https://github.com/chesapeaketechnology/gitlab-templates/blob/release/4/lib/gitlab/ci/templates/jobs/docker/Kaniko.yml#L29) would be best.